lib/converters/v1.0.0/converter-v1-to-v2.js

```diff
@@ -240,8 +240,11 @@ _.assign(Builders.prototype, {
             dataMode = modes[requestV1.dataMode],
             retainEmpty = this.options.retainEmptyValues;
 
-        // set null body if neither data|rawModeData nor dataMode is set.
-        if (!dataMode && _.isEmpty(requestV1.data) && _.isEmpty(requestV1.rawModeData)) {
+        // set null body if neither data|rawModeData nor dataMode is set, or if the dataMode is explicitly set to null
+        // The explicit null check is added to ensure that body is not set in case the dataMode was set to null by the
+        // app, but `data` or `rawModeData` was not empty
+        if ((!dataMode && _.isEmpty(requestV1.data) && _.isEmpty(requestV1.rawModeData)) ||
+            requestV1.dataMode === null) {
```

Comment:
Random note: This is a good expression of intent (but the added tests will pass without this diff too)